pkg/kubectl/swagger_doc.go

```diff
@@ -0,0 +1,170 @@
+				return
+			}
+			pointedModel = allModels[next]
+		} else {
+			fmt.Fprintf(w, "Field: %s doesn't exist.\n", field)
+			return
+		}
+	}
+	printModelInfo(w, pointedModel, apiVer)
+}
+
+func splitDotNotation(model string) (string, []string) {
+	var fieldsPath []string
+	dotModel := strings.Split(model, ".")
+	if len(dotModel) >= 1 {
+		fieldsPath = dotModel[1:]
+	}
+	return dotModel[0], fieldsPath
+}
+
+func getNextModel(prop *swagger.ModelProperty) string {
+	if prop.Ref != nil {
+		return *prop.Ref
+	} else if *prop.Type == "array" && prop.Items.Ref != nil {
+		return *prop.Items.Ref
+	}
+	return ""
+}
+
+func findField(model *swagger.NamedModel, sField string) *swagger.ModelProperty {
+	for _, prop := range model.Model.Properties.List {
+		if prop.Name == sField {
+			return &prop.Property
+		}
+	}
+	return nil
+}
+
+func printModelInfo(w io.Writer, model *swagger.NamedModel, apiVer string) {
+	trimVer := func(str string) string {
+		return strings.TrimPrefix(str, apiVer)
+	}
+
+	fmt.Fprintf(w, "MODEL: %s\n", trimVer(model.Name))
+	fmt.Fprintf(w, "%s\n\n", model.Model.Description)
+
+	fmt.Fprint(w, "\nFIELD DESCRIPTIONS\n")
+	for _, field := range model.Model.Properties.List {
+		fieldType, err := getFieldType(&field.Property, trimVer)
+		if err != nil {
+			fmt.Printf("%v\n", err)
+			os.Exit(1)
+		}
+
+		fmt.Fprintf(w, "-> %s\t<%v>\n", field.Name, fieldType)
+		fmt.Fprintf(w, "\t%s\n", field.Property.Description)
+	}
+	fmt.Fprint(w, "\n")
+}
+
+func getFieldType(prop *swagger.ModelProperty, trimVer func(str string) string) (string, error) {
+	var fieldType string
+
+	if prop.Type == nil {
+		fieldType = trimVer(*prop.Ref)
+	} else if *prop.Type == "any" {
+		// Swagger Spec doesn't return information for maps.
+		fieldType = "map[string]string"
+	} else if *prop.Type == "array" {
+		if prop.Items == nil {
+			return "", fmt.Errorf("Error in swagger spec. Property: %v contains an array without type!", prop)
+		} else if prop.Items.Ref != nil {
+			fieldType = "[]" + trimVer(*prop.Items.Ref)
+		} else if prop.Items.Type != nil {
+			fieldType = "[]" + *prop.Items.Type
+		}
+	} else {
+		fieldType = *prop.Type
+	}
+
+	return fieldType, nil
+}
+
+// SearchForModel prints other models with similar names
+func SearchForModel(inModel, apiVer string, w io.Writer, kubeClient client.Interface) {
+	swaggerSchema, err := kubeClient.SwaggerSchema(apiVer)
+	if err != nil {
+		fmt.Printf("Couldn't read swagger schema from server: %v\n", err)
+		os.Exit(1)
+	}
+
+	trimVer := func(str string) string {
+		return strings.TrimPrefix(str, swaggerSchema.ApiVersion+".")
+	}
+
+	for _, model := range swaggerSchema.Models.List {
+		modelName := trimVer(model.Name)
+		if similarName(modelName, inModel) {
+			fmt.Fprintf(w, "%s\n", modelName)
+		}
+	}
+	fmt.Fprint(w, "\n")
+}
+
+func similarName(modelName, inModel string) bool {
+	mL := strings.ToLower(modelName)
+	iL := strings.ToLower(inModel)
+	return strings.Contains(mL, iL) || strings.Contains(iL, mL)
```

Comment:
I think just the first condition should be sufficient. We should return resources that contain the user specified name. Dont need to return resources whose name is contained in the user specified name.